enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,253 @@
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+metadata:
+  name: "instance"
+spec:
+  logStore:
+    retentionPolicy: 
+      logs.app:
+        maxAge: 7d
+      logs.infra
+        maxAge: 7d
+
+```
+#### Cluster Logging Operator
+The `cluster-logging-operator` will utilize the ClusterLogging CR retention policy to spec
+the desired aforementioned Elasticsearch CR indexManagement.
+#### Elasticsearch API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies:
+    - name: infra-policy
+      phases:
+        hot:
+          actions:
+            rollover:
+              max_size:  30GB         #opinionated value on Elastic recommendations
+              max_age:   3d
+        delete:
+          min_age: 7d
+    templates:
+    - name:  infra
+      template:
+        index_patterns:
+        - node.infra*
+        settings:
+          number_of_shards: 1,        #value based on es node count policy
+          number_of_replicas: 1,      #value based on redundancy policy
+        aliases:
+        - infra  
+    seedings:
+    - prefix: node.infra              #creates node.infra-00001 aliased node.infra-write     
+      policyRef: infra-policy
+```
+#### Elasticsearch Operator
+The `elasticsearch-operator` will be modified to:
+* Expose the index management API
+* Create and seed specified index templates
+* Create the initial indices as needed
+* Block data ingestion if needed until initial index is seeded
+* Deploy curation CronJob for each entry in the specified seedings to rollover using the defined policy
+
+#### Collector 
+* Update the [`fluent-plugin-viaq-data-model`](https://github.com/ViaQ/fluent-plugin-viaq_data_model) to allow defining a static index to write logs
+
+Example configuration:
+```    
+  ....
+   <elasticsearch_index_name>
+     tag "**"
+     name_type static index_name=>'container.app-write'
+   </elasticsearch_index_name>
+```
+#### Deprecations
+* The Curator Cronjob deployed by the `cluster-logging-operator` will be removed. The responsibilities for curation will be subsumed by
+ implementation of Elasticsearch rollover management.
+* Curation configuration by namespace is no longer configurable and is restricted to cluster
+wide settings associated with log type 
+
+## Design Details
+
+### Test Plan
+* Regression tests will be executed to confirm no regressions from previous releases
+#### Unit Tests
+* Unit tests will be modified to account for the change in data design
+#### Integration Tests
+* e2e tests will be modified to account for the change in data design
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+The `elasticsearch-operator` will migrate existing indices to the new data design:
+* Indices beginning with `project.*` are indexed to `container.app-write`
+* Indices beginning with `.operations.*` are indexed to `journal.infra-write`
+* Migrated indices are deleted after migration
```

Comment:
How does the migration process handle the situation where the retention policy is changed along with the upgrade?

Reply:
> How does the migration process handle the situation where the retention policy is changed along with the upgrade?
The data will be subject to the new retention policy.  It's possible we could infer their policy from the previous curator deployment but I don't know how that manifests itself into the CL instance since the operator is not supposed to modify the spec.  @bparees ?

Reply:
reducing their retention on upgrade(and thus immediately throwing away some set of logs) seems like a non-starter.

Reply:
So, if I have a lower retention policy in the new cluster, when the data is migrated, is it immediately deleted?  If it is not, is that a problem?

Reply:
We could additionally handle old data retention and curation by simply doing nothing, including not migrating data to the new schema.  We would leave the existing cronjob and ES would be optimized over time as old data is curated away to the point where data in the old schema is no longer present.  This would be an easier path for us to follow and addresses @portante concerns.  This will work as long as curator does not match any indices from the new schema which I believe it does not.

Reply:
@bparees @sichvoge any reason not to simply skip data migration and allow curation to curate them out. it would mean we we have to make adjustments to view the data but we would avoid the need to move data.  most of our dedicated clusters restrict data to 7 days now because of scale issues

Reply:
is curation on by default?  or might some clusters not have any curation happening?

if we can tolerate both data schemas from a query/view perspective, indefinitely, then this seems like an ideal answer regardless...

Reply:
Landed on leaving aliases alone and aliasing them to the new format, relying on DSL for security, and curation to roll off.

Reply:
> Landed on leaving aliases alone and aliasing them to the new format, relying on DSL for security, and curation to roll off.

leaving **pre-existing indicies** alone and adding them to the new alias format.